<?xml version="1.0"?>
<launch>
    <arg name="publisher_rate" default="10"/>
    <arg name="x" default="15"/>
    <arg name="y" default="30"/>
    <arg name="z" default="0"/>
    <arg name="roll" default="2"/>
    <arg name="pitch" default="3"/>
    <arg name="yaw" default="4"/>
    <arg name="parent_frame" default="/world"/>
    <arg name="child_frame" default="/talk"/>
    <arg name="record_rosbag" default="false" />
    <arg name="rosbag_directory" default="bag"/>

    <group>
        <node pkg="beginner__tutorials" type="talker_node" name = "talker">
            <param name="publisher_rate" value="$(arg publisher_rate)"/>
            <param name="x" value="$(arg x)" type = "double"/>
            <param name="y" value="$(arg y)" type = "double"/>
            <param name="z" value="$(arg z)" type = "double"/>
            <param name="roll" value="$(arg roll)" type = "double"/>
            <param name="pitch" value="$(arg pitch)" type = "double"/>
            <param name="yaw" value="$(arg yaw)" type = "double"/>
            <param name="parent_frame" value="$(arg parent_frame)"/>
            <param name="child_frame" value="$(arg child_frame)"/>
        </node>
    </group>

    <group>
        <node pkg="beginner__tutorials" type="listener_node" name = "listener" output="screen">
        </node>
    </group>

    <group if="$(arg record_rosbag)">
        <node pkg="rosbag" type="record" name="record"
            args="record -o $(arg rosbag_directory) -a"/> 
    </group>

</launch>